OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -30,7 +30,7 @@ public Infiltrate(Actor self, Actor target)
 
 		protected override void OnInside(Actor self)
 		{
-			if (target.IsDead || target.Owner == self.Owner)
+			if (target.IsDead || target.Owner.IsAlliedWith(self.Owner))
```

Comment:
This should be removed entirely IMO as the Order ought to take care of proper targeting.